mirdata/datasets/billboard.py

```diff
@@ -0,0 +1,480 @@
+                        if len(chords):
+                            event["chords"] = chords
+                        else:
+                            event["notes"].append(i)
+                    o["events"].append(event)
+        return o
+
+    o = parse(s)
+    return o
+
+
+def _timed_sections(parsed: Dict) -> List:
+    """
+    Author:
+        Brian Whitman
+        brian@echonest.com
+        https://gist.github.com/bwhitman/11453443
+    Given a salami parse return a list of parsed chords with timestamps & deltas
+    """
+    timed_sections = []
+    tic = 0
+    for i, e in enumerate(parsed["events"]):
+        sections = []
+        try:
+            dt = parsed["events"][i + 1]["time"] - e["time"]
+        except IndexError:
+            dt = 0
+
+        section = None
+        if e.get("notes"):
+            if len(e.get("notes")) > 1:
+                section = (e.get("notes")[0], e.get("notes")[1])
+            sections.append(section)
+
+        tic = e["time"]
+        if len(sections):
+            seconds_per_chord = dt / float(len(sections))
+            for c in sections:
+                timed_sections.append(
+                    {"time": tic, "section": c, "length": seconds_per_chord}
+                )
+                tic = tic + seconds_per_chord
+    return timed_sections
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The McGill Billboard dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="billboard",
+            track_class=Track,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.cached_property
+    def _metadata(self):
+        metadata_path = os.path.join(self.data_home, "billboard-2.0-index.csv")
+
+        if not os.path.exists(metadata_path):
+            raise FileNotFoundError("Metadata not found. Did you run .download()?")
+
+        with open(metadata_path, "r") as fhandle:
+            reader = csv.reader(fhandle, delimiter=",")
+            next(reader, None)
+            raw_data = []
+            for line in reader:
+                if line != []:
+                    raw_data.append(line)
+
+        metadata_index = {}
+        for line in raw_data:
+            track_id = line[0]
+            metadata_index[track_id] = {
+                "chart_date": line[1],
+                "target_rank": int(line[2]) if line[2] else None,
+                "actual_rank": int(line[3]) if line[3] else None,
+                "title": line[4],
+                "artist": line[5],
+                "peak_rank": int(line[6]) if line[6] else None,
+                "weeks_on_chart": int(line[7]) if line[7] else None,
+            }
+        return metadata_index
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
```

Comment:
Add `load_sections` and `load_chords` here as well

Reply:
Fixed in https://github.com/Spijkervet/mirdata/commit/fa6e09e34e7c2b8f9a70e4e4fcaedb5da80d4ac8